sklearn/manifold/_mds.py

```diff
@@ -306,6 +308,75 @@ def smacof(
         return best_pos, best_stress
 
 
+def eigh_scaler(dissimilarities, n_components=2):
+    """
+    Computes multidimensional scaling using eigh solver.
+
+    Parameters
+    ----------
+    dissimilarities : ndarray, shape (n_samples, n_samples)
+        Pairwise dissimilarities between the points. Must be euclidean.
+    n_components : int, optional, default=2
+        Number of dimension in which to immerse the dissimilarities.
+
+    Returns
+    ----------
+    embedding : ndarray, shape (n_samples, n_components)
+        Coordinates of the points in a ``n_components``-space.
+
+    stress : float
+        The final value of the stress (sum of squared distance of the
+        disparities and the distances for all constrained points).
+
+    References
+    ----------
+    "An Introduction to MDS" Florian Wickelmaier
+    Sound Quality Research Unit, Aalborg University, Denmark (2003)
+
+    "Metric and Euclidean properties of dissimilarity coefficients"
+    J. C. Gower and P. Legendre, Journal of Classification 3, 5–48 (1986)
+
+    """
+
+    dissimilarities = check_symmetric(dissimilarities, raise_exception=True)
+
+    n_samples = dissimilarities.shape[0]
+
+    # Centering matrix
+    J = np.eye(*dissimilarities.shape) - (1.0 / n_samples) * (
+        np.ones(dissimilarities.shape)
+    )
+
+    # Double centered matrix
+    B = -0.5 * np.dot(J, np.dot(dissimilarities**2, J))
```

Comment:
Use of broadcasting would be much more efficient here!

Reply:
Sorry for such a late reply, but could you explain a bit how I could employ broadcasting here?

Reply:
Centering by columns and by rows can be done like this instead of using a centering matrix:

```python
B = dissimilarities ** 2
B = B - np.mean(B, axis=0)
B = B - np.mean(B, axis=1, keepdims=True)
```

Reply:
Thank you for the advice!